monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -303,22 +366,39 @@ class BundleGen(AlgoGen):
         python -m monai.apps.auto3dseg BundleGen generate --data_stats_filename="../algorithms/data_stats.yaml"
     """
 
-    def __init__(self, algo_path: str = ".", algos=None, data_stats_filename=None, data_src_cfg_name=None):
-        self.algos: Any = []
-
-        if algos is None or isinstance(algos, str):
-            # trigger the download process
-            zip_download_dir = TemporaryDirectory()
-            algo_compressed_file = os.path.join(zip_download_dir.name, "algo_templates.tar.gz")
-            download_and_extract(default_algo_zip if algos is None else algos, algo_compressed_file, algo_path)
-            zip_download_dir.cleanup()
-            sys.path.insert(0, os.path.join(algo_path, "algorithm_templates"))
-            algos = deepcopy(default_algos)
-            for name in algos:
-                algos[name]["template_path"] = os.path.join(
-                    algo_path, "algorithm_templates", algos[name]["template_path"]
-                )
+    def __init__(
+        self,
+        algo_path: str = ".",
+        algos: Optional[Union[Dict, List, str]] = None,
+        templates_path_or_url: Optional[str] = None,
```

Comment:
it's probably not introduced by this PR, but `algo_path`, `algos` and `template_path_or_url` are still quite complicated. Shall we create a separate set APIs, for example, `install_algo_templates` or `prepare_algo_templates` or something to manage a set of candidate templates? cc @mingxin-zheng

Reply:
I haven't given much thoughts here because I am in another rabbit hole. In my mind, what makes bundleGen complicated is the instantiation of the algorithms. The function name "install_algo_templates" sounds like the instantiation could be taken out, which I think could be a good idea. Is that what you're thinking @wyli ?

Reply:
yes, `install_algo_templates` should be decoupled from `BundleGen` (this could be download or copy folder). Then `BundleGen` can assume that a set of templates are ready to use locally.

Reply:
Let's continue in a  separate issue for this , https://github.com/Project-MONAI/MONAI/issues/5631
but since, a tentative plan is to move algo templates into core in the future, we won't need to download any zip files, and the API will be simpler